### PR TITLE
bug fix

### PR DESCRIPTION
**Pull request by Dhiraj100892**
## Motivation and Context

- There is transformation issue in `go_to_absolute` function in `habitat` wrapper
- Currently `pix_to_3dpt` & `get_current_pcd` function in `habitat` wrapper returns point in global frame while in `locobot`, it returns it in robot base frame 

## How Has This Been Tested

- For `go_to_absolute` function, it has been tested using the testcase
- For `pix_to_3dpt` & `get_current_pcd` has been tested based on the visualization

## Types of changes

- [ ] Docs change / refactoring / dependency upgrade
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have completed my CLA (see **CONTRIBUTING**)
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: src/pyrobot/habitat/camera.py [hunk 1]
```diff
@@ -112,15 +111,14 @@ def pix_to_3dpt(self, rs, cs, in_cam=False, initial_rotation=None):
...
-
     def _cam2pyrobot(self, pts, initial_rotation=None):
         """
         here, points are  given in camera frame
         the thing to do next is to transform the points from camera frame into the
         global frame of pyrobot environment
         This does not translate to what habitat thinks as origin,
         because pyrobot's habitat-reference origin is `self.agent.init_state`
-        So, CAMERA frame -> HABITAT frame -> PYROBOT frame
+        So, CAMERA frame -> HABITAT frame -> PYROBOT frame (robot base frame)
```
Comment (soumith) [inline src/pyrobot/habitat/camera.py:121]: is this correct?

@kalyanvasudev said (when we were debugging last time) that PYROBOT frame is basically the global coordinate frame in the pyrobot environment.

The line above this also describes it differently from robot base frame:

```
the thing to do next is to transform the points from camera frame into the
        global frame of pyrobot environment
```

**Comment by soumith** [comment]
> Currently pix_to_3dpt & get_current_pcd function in habitat wrapper returns point in global frame while in locobot, it returns it in robot base frame

Ok, yea then when we were implementing it for Habitat, we completely had the wrong idea.

How does one then transform the point cloud into the global frame after it is returned by `get_current_pcd`?

File: src/pyrobot/habitat/camera.py (hunk 1 above)
Comment (kalyanvasudev) [reply src/pyrobot/habitat/camera.py:121]: Yeah, this seems right. Basically, on the real robot, the points are transformed into whatever frame set here - https://github.com/facebookresearch/pyrobot/blob/Develop/src/pyrobot/cfg/locobot_config.py#L125 which is `/base_link` i.e robot base frame.

**Comment by kalyanvasudev** [comment]
> > Currently pix_to_3dpt & get_current_pcd function in habitat wrapper returns point in global frame while in locobot, it returns it in robot base frame
> 
> Ok, yea then when we were implementing it for Habitat, we completely had the wrong idea.
> 
> How does one then transform the point cloud into the global frame after it is returned by `get_current_pcd`?

This should be pretty straightforward, we get the robot base pose from `bot.base.get_state("odom")`. We can then use it to construct a transformation matrix that takes the points from robot base frame to the world frame.

**Pull request merged**